enhancements/sdk-migrate-bundle.md

```diff
@@ -0,0 +1,379 @@
+   - if we do build images, we should add image builder flag
+1. What will happen to the existing command in SDK for deploying an operator
+   packagemanifests using OLM? Do we still want to support that and will it stay intact?
+   - we will deprecate the `run packagemanifests` command and remove in SDK 2.0
+
+## Summary
+
+Add a subcommand to the Operator SDK CLI that will allow users to migrate
+their operator's packagemanifest format to the new bundle format.
+
+## Motivation
+
+There are operators that exist that are still using the packagemanifest
+format. This subcommand will allow them to migrate these packagemanifests to
+bundles which is the preferred OLM packaging format.
+
+### Goals
+
+- Provide a subcommand to migrate one packagemanifest to a set of bundles.
+- Deprecate `packagemanifests` features:
+  - Deprecate `run packagemanifests`
+  - Deprecate `generate packagemanifests`
+
+### Non-Goals
+
+- Solve the world?
+- There will be no migration from bundles to packagemanifests
+
+## Proposal
+
+### User Stories [optional]
+
+#### Story 1
+
+As an operator developer, I would like to migrate my operator which uses
+the packagemanifest format to the new bundle format.
+
+#### Story 2
+
+As an catalog admin for my operators, I would like to be able to migrate a
+collection of packagemanifests to bundles.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The Operator SDK currently supports generating & validating
+packagemanifests & bundles. This enhancement will add a new command, `migrate
+bundle` to the CLI. This new command will migrate an existing packagemanifest to
+a bundle.
+
+More information about the different format can be found at the URLs below:
+
+- Operator bundles: https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md
+- Package manifests format: https://github.com/operator-framework/operator-registry/tree/v1.5.3/#manifest-format
+
+The Operator SDK will traverse the given packagemanifest directory and for each
+of the versions create a new bundle.
+
+#### Generating bundle on disk
+
+In order to generate bundles to disk, we will refactor the `generate bundle`
+logic to be reusable in generating the bundles.
+
+===================== REWRITE BELOW =======================
+The `GenerateFunc` takes in the following arguments:
+
+- directory string - local directory where the manifests and metadata are
+  located
+- outputDir string - where manifests and metadata are copied
+- packageName string - the name of the package that bundle image belongs to
+- channels string - the list of channels that bundle image belongs to
+- channelDefault string - the default channel for the bundle image
+
+In order to use this function, we will obtain the required parameters from the
+listed *source*:
+
+| parameter | source |
+| --------- | ------ |
+| directory | packagemanifest directory |
+| outputDir | default `bundle` dir or flag `--output-dir` |
```

Comment:
I prefer only --output and then keep in the flag description that is expected the path of a dir. 
If not informed where the output will occur? Are we covering these details in this EP?